examples/car_control/run.py

```diff
@@ -0,0 +1,136 @@
+
+import argparse
+import numpy as np
+import os
+import spear
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    args = parser.parse_args()
+
+    np.set_printoptions(linewidth=200)
+
+    # load config
+    config = spear.get_config(user_config_files=[os.path.realpath(os.path.join(os.path.dirname(__file__), "user_config.yaml"))])
+    spear.configure_system(config)
+    instance = spear.Instance(config)
+
+    with instance.begin_frame():
+        gameplay_statics_uclass = instance.unreal_service.get_static_class(class_name="UGameplayStatics")
+        controller_uclass = instance.unreal_service.load_class(class_name="UObject", outer=0, name="/Script/Engine.Controller")
+        bp_car_uclass = instance.unreal_service.load_class(class_name="UObject", outer=0, name="/Game/VehicleTemplate/Blueprints/OffroadCar/OffroadCar_Pawn.OffroadCar_Pawn_C")
+        vehicle_movement_component_class = instance.unreal_service.load_class(class_name="UObject", outer=0, name="/Script/ChaosVehicles.ChaosVehicleMovementComponent")
+
+        get_player_controller_func = instance.unreal_service.find_function_by_name(uclass=gameplay_statics_uclass, function_name="GetPlayerController")
+        set_game_paused_func = instance.unreal_service.find_function_by_name(uclass=gameplay_statics_uclass, function_name="SetGamePaused")
+        possess_func = instance.unreal_service.find_function_by_name(uclass=controller_uclass, function_name="Possess")
+        set_throttle_input_func = instance.unreal_service.find_function_by_name(uclass=vehicle_movement_component_class, function_name="SetThrottleInput")
+
+        gameplay_statics_default_object = instance.unreal_service.get_default_object(uclass=gameplay_statics_uclass, create_if_needed=False)
+        player_controller = spear.func_utils.to_handle(instance.unreal_service.call_function(uobject=gameplay_statics_default_object, ufunction=get_player_controller_func,
+                                                                                             args={"PlayerIndex": 0})["ReturnValue"])
+
+        bp_car_actor = instance.unreal_service.spawn_actor_from_uclass(
+            uclass=bp_car_uclass,
+            location={"X": 0.0, "Y": 0.0, "Z": 0.0}, rotation={"Roll": 0.0, "Pitch": 0.0, "Yaw": 0.0},
+            spawn_parameters={"Name": "Agent", "SpawnCollisionHandlingOverride": "AlwaysSpawn"}
+        )
+        vehicle_movement_component = instance.unreal_service.get_component_by_class(actor=bp_car_actor, uclass=vehicle_movement_component_class)
+
+        # need player_controller current actor to inject input
+        instance.unreal_service.call_function(
+            uobject=player_controller,
+            ufunction=possess_func,
+            args={"InPawn": spear.func_utils.to_ptr(bp_car_actor)})
+
+    with instance.end_frame():
+        pass
+
+    for _ in range(100):
```

Comment:
Is it intentional that these for loops are sometimes calling `UFUNCTIONS` and sometimes going through the input system?

Reply:
updated. `IA_Steering` and `set_throttle_input_func` both update actor state and only need to call when change needed.